src/utils/order_state_utils.ts

```diff
@@ -60,13 +60,27 @@ export class OrderStateUtils {
         );
         const filledTakerTokenAmount = await this.exchangeWrapper.getFilledTakerAmountAsync(orderHash, methodOpts);
         const canceledTakerTokenAmount = await this.exchangeWrapper.getCanceledTakerAmountAsync(orderHash, methodOpts);
+        const unavailableTakerTokenAmount =
+          await this.exchangeWrapper.getUnavailableTakerAmountAsync(orderHash, methodOpts);
+        const totalMakerTokenAmount = signedOrder.makerTokenAmount;
+        const totalTakerTokenAmount = signedOrder.takerTokenAmount;
+        const remainingTakerTokenAmount = totalTakerTokenAmount.minus(unavailableTakerTokenAmount);
+        // 200 in order, 100 unavailable  = 100 remaning, 0.5 remaining in taker proportion
+        const remainingTakerProportion = remainingTakerTokenAmount.dividedBy(totalTakerTokenAmount);
+        const remainingMakerTokenAmount = remainingTakerProportion.times(totalMakerTokenAmount);
+        // min allowance, balance in account of maker
+        const fillableMakerTokenAmount = BigNumber.min([makerProxyAllowance, makerBalance]);
+        // min ^, remaining order maker token amount
+        const remainingFillableMakerTokenAmount = BigNumber.min(fillableMakerTokenAmount, remainingMakerTokenAmount);
+        // edge case when maker token is ZRX
```

Comment:
Let's prefix this comment with `TODO:` and have it be a call to action: `Handle edge case where...`